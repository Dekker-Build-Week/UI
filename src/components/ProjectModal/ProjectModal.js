import React from 'react';
import Modal from "@material-ui/core/Modal";
import Backdrop from "@material-ui/core/Backdrop";

import { withStyles } from "@material-ui/core/styles";
import projectModalStyle from "assets/jss/material-dashboard-react/components/projectModalStyle.js";

class ProjectModal extends React.Component {
    constructor(props) {
        super(props);

        this.handleClose = this.handleClose.bind(this);
    }

    componentWillReceiveProps(newProps) {
        this.setState({ isOpen : newProps.modalOpen })
    }

    handleClose() {
        this.props.handleClose();
    }
    
    render() {
        const { classes, children } = this.props;

        return (
                <Modal
                    aria-labelledby="spring-modal-title"
                    aria-describedby="spring-modal-description"
                    className={classes.modalBody}
                    open={this.props.modalOpen}
                    onClose={this.handleClose}
                    BackdropComponent={Backdrop}
                >
                  {children}
                </Modal>
        )
    }
}
export default withStyles(projectModalStyle)(ProjectModal);