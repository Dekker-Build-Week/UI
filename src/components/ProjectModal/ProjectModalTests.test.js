import React from 'react';
import ProjectModal from './ProjectModal';

import renderer from 'react-test-renderer';

describe('Project Modal Test Suite', () => {
    it('Renders the ProjectModal successfully', () => {
        const tree = renderer 
        .create(<ProjectModal><p>Hey</p></ProjectModal>)
            .toJSON();

        expect(tree).toMatchSnapshot();     
    })
})