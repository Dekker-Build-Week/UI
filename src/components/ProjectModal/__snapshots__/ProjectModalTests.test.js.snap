// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Modal Test Suite Renders the ProjectModal successfully 1`] = `null`;
